sycl/doc/Assert.md

```diff
@@ -0,0 +1,331 @@
+# Assert feature
+
+**IMPORTANT**: This document is a draft.
+
+Using the standard C++ `assert` API ("assertions") is an important debugging
+technique widely used by developers. This document describes the design of
+supporting assertions within SYCL device code.
+The basic approach we chose is delivering device-side assertions as host-side
+asynchronous exceptions, which allows further extensibility, such as better
+error handling or potential recovery.
+
+As usual, device-side assertions can be disabled by defining `NDEBUG` macro at
+compile time.
+
+## Use-case example
+
+```
+using namespace sycl;
```

Comment:
Suggest adding the include files also to illustrate that the application needs to include `<cassert>` too:

```
#include <cassert>
#include <sycl/sycl.hpp>
```

Reply:
Added